web/src/recent_view_data.ts

```diff
@@ -1,26 +1,33 @@
 import * as people from "./people";
 import {get_key_from_message} from "./recent_view_util";
+import type {Message} from "./types";
 
-export const topics = new Map();
+export type TopicData = {
+    last_msg_id: number;
+    participated: boolean;
+    type: "private" | "stream";
+};
+export const topics = new Map<string | undefined, TopicData>();
```

Comment:
@amanagr do you understand why `undefined` is a possible key here? I'm not sure what situation that would occur in.

Reply:
Opened #26885, `undefined` should not be possible there.